File: src/main/java/com/karan/restaurant/review/system/service/UserService.java
package com.karan.restaurant.review.system.service;

import com.karan.restaurant.review.system.dto.UserRequestDTO;
import com.karan.restaurant.review.system.dto.UserResponseDTO;

import java.util.List;
public interface UserService {
    // create a new user
    public UserResponseDTO createUser(UserRequestDTO userRequestDTO);
    // get user by id
    public UserResponseDTO getUserById(Long id);
    // get all users
    public List<UserResponseDTO> getAllUsers();
    // update user by id
    public UserResponseDTO updateUser(Long id, UserRequestDTO userRequestDT0);
    // delete user by id
    public void deleteUser(Long id);
}

File: frontend/src/services/userService.js
import api from '../utils/api';

export const userService = {
  // Create user
  createUser: async (userData) => {
    const response = await api.post('/v1/users/admin/register', userData);
    return response.data;
  },

  // Get all users
  getAllUsers: async () => {
    const response = await api.get('/v1/users');
    return response.data;
  },

  // Get user by ID
  getUserById: async (id) => {
    const response = await api.get(`/v1/users/${id}`);
    return response.data;
  },

  // Update user
  updateUser: async (id, userData) => {
    const response = await api.put(`/v1/users/${id}`, userData);
    return response.data;
  },

  // Delete user
  deleteUser: async (id) => {
    await api.delete(`/v1/users/${id}`);
  },
};